examples/platforms/posix/radio.c

```diff
@@ -715,7 +714,13 @@ void radioSendAck(void)
     sAckFrame.mLength    = IEEE802154_ACK_LENGTH;
     sAckMessage.mPsdu[0] = IEEE802154_FRAME_TYPE_ACK;
 
-    if (isDataRequestAndHasFramePending(&sReceiveFrame))
+    if (
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2
+        (otMacFrameIsData(&sReceiveFrame) || otMacFrameIsDataRequest(&sReceiveFrame))
```

Comment:
Agree that it is not strictly required by the specification, but useful to not set the Frame Pending bit in ACKs sent in response to Data frames.

Reply:
I chose the current implementation to make the code simpler, otherwise, the radio driver must know the peer's Thread version in the context of interrupt.
We could use something like the src match table for this, or verify the frame version is 2015.
However, I prefer the simpler solution as Thread 1.1 explicitly stated to ignore the bit. Thoughts?

Reply:
Please check the below comments. At least devices shipped with OT1.1 in current code will not ignore the Frame Pending bit in ACKs not in response to Data Request frames.

Reply:
If current 1.1 code doesn't ignore, then it's a bug. I think we should fix the bug as well.